corehq/tabs/tabclasses.py

```diff
@@ -159,10 +159,27 @@ def _get_tools_items(self):
             'icon': 'icon-tasks fa fa-tasks',
             'show_in_dropdown': True,
         }]
-        if toggles.USER_CONFIGURABLE_REPORTS.enabled(self.couch_user.username):
-            # Only show for **users** with the flag. This flag is also available for domains
-            # but should not be granted by domain, as the feature is too advanced to turn
-            # on for all of a domain's users.
+        from corehq.apps.users.models import Permissions
+        # For SaaS envs, if we want to display the tab only if Feature Flag is
+        # enabled on domain and the role has the permission to edit_ucrs
```

Comment:
What's the rationale behind limiting this change to SaaS environments?  An easier rollout?  Seems like the kind of control anyone who uses the UCR feature flag might want.

I'm not crazy about using `IS_SAAS_ENVIRONMENT` for this purpose, as it seems outside the scope of what it's for (allowing users to sign-up and create their own projects independently).  I'm concerned this sort of thing would encourage third parties to turn on the `IS_SAAS_ENVIRONMENT` flag to gain access to features like this when something more granular would be more appropriate.

Reply:
Hey Ethan,

An easier rollout was the only motivation behind adding the check for `IS_SAAS_ENVIRONMENT`. 

How would you suggest that this should be approached?

Reply:
If you do end up writing a migration for it as Jenny brought up in another thread, then I think the change would be seamless, right?  I'd imagine you could then just turn it on for all environments.

Reply:
Yeah. We can. Thanks for your inputs @esoergel. I will write migration for it then.

Reply:
Thanks @AmitPhulera , that'd be great